tensorflow/compiler/mlir/tosa/transforms/legalize_utils.cc

```diff
@@ -549,18 +549,6 @@ LogicalResult ApplyPatternsWithShapeResolution(
     return failure();
   }
 
-  // Check that constant attributes types and op types match up. If the lowering
-  // needs to change a type (e.g. fp16 -> fp32) its possible the return type
-  // could be incorrect.
-  //
-  // This should be investigate for whether it is still necessary due to quant
-  // type stripping changing.
-  func.walk([&](tosa::ConstOp op) {
```

Comment:
We should not be removing this cast. As part of type conversion the attribute element type may change during legalization, this can result in the attribute type no longer matching the result type. If we want to maintain the quantization information we need to then preserve the quantized type with the new element type, not just remove this pass.

If we remove this section entirely we will break parts of the pipeline.